Bottom: 06c42c7955064a64ddd9127de1440c8ec968fd76
Top:    66cb3f35b7e4ca2e0294337a1e2345d056269d41
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-26 00:47:57 +0100

Refresh of draw-c-split-gathering-routine-2

---

diff --git a/src/draw.c b/src/draw.c
index 8cf2bf4..5af9fd1 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -202,9 +202,8 @@ element_callback (const BoxType * b, void *cl)
 static void
 draw_element_name (ElementType *element)
 {
-  if (gui->gui && TEST_FLAG (HIDENAMESFLAG, PCB))
-    return;
-  if (TEST_FLAG (HIDENAMEFLAG, element))
+  if ((TEST_FLAG (HIDENAMESFLAG, PCB) && gui->gui) ||
+      TEST_FLAG (HIDENAMEFLAG, element))
     return;
   if (doing_pinout || doing_assy)
     gui->set_color (Output.fgGC, PCB->ElementColor);
@@ -409,17 +408,6 @@ draw_element_package (ElementType *element)
   END_LOOP;
 }
 
-static int
-element_callback (const BoxType * b, void *cl)
-{
-  ElementTypePtr element = (ElementTypePtr) b;
-  int *side = cl;
-
-  if (ON_SIDE (element, *side))
-    draw_element_package (element);
-  return 1;
-}
-
 /* ---------------------------------------------------------------------------
  * prints assembly drawing.
  */
@@ -1484,7 +1472,7 @@ DrawElementName (ElementTypePtr Element)
 
   if (TEST_FLAG (HIDENAMEFLAG, Element))
     return;
-  AddPart (&ELEMENT_TEXT (PCB, Element));
+  DrawTextLowLevel (&ELEMENT_TEXT (PCB, Element), 0);
 }
 
 /* ---------------------------------------------------------------------------
